include/PR/os_internal.h

```diff
@@ -1,14 +1,24 @@
 #ifndef PR_OS_INTERNAL_H
 #define PR_OS_INTERNAL_H
 
+#ifdef _LANGUAGE_C
+
 #include "ultratypes.h"
-#include "os_message.h"
-#include "os_pi.h"
-#include "os_internal_rsp.h"
 
 typedef struct __osHwInt {
     /* 0x0 */ s32 (*handler)(void);
     /* 0x4 */ void* stackEnd;
 } __osHwInt; // size = 0x8
 
+#else
+
+/* __osHwInt struct member offsets */
+#define HWINT_CALLBACK 0x00
+#define HWINT_SP       0x04
+
+/* __osHwInt struct size */
+#define HWINT_SIZE     0x8
```

Comment:
Could be worth having `static_assert`s to make sure those values don't get desynced?
Also having an extra zero in some and not in the other is a bit weird (`0x04` vs `0x8`)